nik4.py

```diff
@@ -621,6 +623,18 @@ def run(options):
     parser.add_argument('output', help='Resulting image file')
     options = parser.parse_args()
 
+    if options.tiles:
+        if re.search(r'^\d+$', options.tiles):
+            options.tiles_x = int(options.tiles)
+            options.tiles_y = options.tiles_x
+        else:
+            match = re.search(r'^(\d+)x(\d+)$', options.tiles)
+            if (match):
+               options.tiles_x = int(match.group(1))
+               options.tiles_y = int(match.group(2))
+        if options.tiles_x == 0 or options.tiles_y == 0:
```

Comment:
I think both checks should go here, that `x` and `y ` are between 1 and 12, like the old code suggested. Maybe in form `1 <= tiles_x * tiles_y <= 144`.